### PR TITLE
FRO 156 fam doc updates

### PR DESCRIPTION
**Pull request by PatFawbertMills**
- Supporting new object models and hierarchy for family and document
- Applying new search context & caching technique for highlighting and document matches
- New document view page with passage highlighting
- Targets integrated into geo page and family page
- Adding data attributes for analytics
- Lots of new typings

### CODE REVIEW

File: src/components/Targets.tsx [hunk 1]
```diff
@@ -1,23 +1,42 @@
...
+  };
+
   return (
-    <ul className="ml-4 list-disc list-outside">
+    <ul className="ml-4 list-disc list-outside" data-cy="targets">
       {targets.map((target) => (
-        <li className="mb-4" key={target.target}>
-          <span className="text-blue-700">{target.target}</span>
+        <li className="mb-4" key={target.ID}>
+          <span className="text-indigo-700" dangerouslySetInnerHTML={{ __html: target.Description }} />
```
Comment (diversemix) [inline src/components/Targets.tsx:22]: Bit of a smell defining a color here, no? :nose:

File: src/components/blocks/Event.tsx [hunk 2]
```diff
@@ -28,13 +28,17 @@ const Event = ({ event, last, index }: EventProps) => {
   return (
     <div className={`text-center w-[140px] relative flex-shrink-0`}>
       <div className={`h-[2px] bg-blue-600 absolute top-1/2 translate-y-[-1px] z-0 ${timelineStyles}`} />
-      <div className="flex items-end justify-center h-[100px]">{!even && renderText(name, month + " " + year)}</div>
+      <div className="flex items-end justify-center h-[100px]">{!even && renderText(title, month + " " + year)}</div>
       <div className="flex place-content-center h-full relative z-10">
         <div className="circle-container">
-          {category ? <div className="circle-icon">{getCategoryIcon(category, "22")}</div> : <div className={index === 0 || last ? "circle-full" : "circle-empty"}></div>}
+          {/* {category ? (
```
Comment (diversemix) [inline src/components/blocks/Event.tsx:34]: The commented code intentional?

File: src/components/blocks/List.tsx [hunk 3]
```diff
@@ -6,13 +6,13 @@ const List = ({ list, bulleted = false }) => {
...
-        <li key={`listitem-${index}-${item.name}`} className={bulleted ? "" : "inline"}>
+        <li key={`listitem-${index}-${item}`} className={bulleted ? "" : "inline"}>
           {
             <>
-              {index > 0 && !bulleted && ", "} {item.name}
+              {index > 0 && !bulleted && ", "} {item}
             </>
           }
-          {item?.children && renderChildren(item.children)}
+          {/* {item?.children && renderChildren(item.children)} */}
```
Comment (diversemix) [inline src/components/blocks/List.tsx:15]: Commented code

File: src/components/document/FamilyDocument.tsx [hunk 4]
```diff
@@ -1,56 +1,56 @@
...
+        <div className="flex-1 flex flex-wrap gap-x-8 items-center">
+          {!!content_type && <span>{renderContentType(content_type)}</span>}
+          <span className="capitalize">{variant.toLowerCase()}</span>
+          <span>{year}</span>
+          {!canView && <span>Document preview is not currently available</span>}
         </div>
         {hasMatches && (
           <div className="flex-0">
-            <MatchesButton dataAttribute={slug} count={matches} overideText={matches === 0 ? "view Document" : null} />
+            <MatchesButton dataAttribute={slugs[0]} count={matches} overideText={matches === 0 ? "view Document" : null} />
```
Comment (diversemix) [inline src/components/document/FamilyDocument.tsx:53]: We should probably ensure that the backend orders the slugs to the most recent is first :thinking:

File: src/components/headers/DocumentSlideout.tsx [hunk 5]
```diff
@@ -43,7 +43,7 @@ const DocumentSlideout = ({ document, searchTerm, showPDF, setShowPDF }: TProps)
                 {searchTerm}"
               </h3>
             </div>
-            <DocumentMenu document={document} />
+            {/* <DocumentMenu document={document} /> */}
```
Comment (diversemix) [inline src/components/headers/DocumentSlideout.tsx:46]: Commented code

File: src/pages/document/[id].tsx [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+    const { data: returnedData } = await client.get(`/documents/${id}`, { group_documents: true });
+    familyData = returnedData;
+  } catch (error) {
+    // TODO: handle error more elegantly
+  }
+
+  if (familyData) {
+    try {
+      const targetsRaw = await axios.get<TTarget[]>(
+        `https://cpr-staging-targets-json-store.s3.eu-west-1.amazonaws.com/families/${familyData.import_id}.json`
```
Comment (diversemix) [inline src/pages/document/[id].tsx:326]: Hardcoded to staging?

File: src/pages/geographies/[id].tsx [hunk 7]
```diff
@@ -268,11 +262,26 @@ export const getServerSideProps: GetServerSideProps = async (context) => {
...
   }
+  try {
+    let countries: TGeography[] = [];
+    const configData = await client.get(`/config`, null);
+    const response_geo = extractNestedData<TGeography>(configData.data?.metadata?.CCLW?.geographies, 2, "");
+    countries = response_geo.level2;
+    const country = getCountryCode(id as string, countries);
+    if (country) {
+      const targetsRaw = await axios.get<TTarget[]>(
+        `https://cpr-staging-targets-json-store.s3.eu-west-1.amazonaws.com/geographies/${country.toLowerCase()}.json`
```
Comment (diversemix) [inline src/pages/geographies/[id].tsx:278]: Hardcoded s3 url

File: src/pages/litigation/[caseId].tsx [hunk 8]
```diff
@@ -261,9 +261,9 @@ const LitigationCoverPage = () => {
                   <DocumentInfo id="filed-tt" heading="Filed" text="File by, Party Type" />
                   <DocumentInfo id="responding-tt" heading="Responding" text="Responding party, Type" />
                   <DocumentInfo id="intervenor-tt" heading="Intervenor" text="Intervenor, Type" />
-                  {1 > 0 && <DocumentInfo id="keywords-tt" heading="UNFCCC pillars" list={UNFCCCPillars} />}
+                  {/* {1 > 0 && <DocumentInfo id="keywords-tt" heading="UNFCCC pillars" list={UNFCCCPillars} />}
```
Comment (diversemix) [inline src/pages/litigation/[caseId].tsx:264]: Commented out code

File: src/utils/searchCache.ts [hunk 9]
```diff
@@ -31,7 +31,16 @@ const saveCache = (newCache: TCacheSearch) => {
   window.localStorage.setItem(CACHE_NAME, JSON.stringify(newCache));
 };
 
+const clearOldCache = () => {
+  const cache = getCache();
+  const newCache = {
+    cache: cache.cache.filter((search) => search.timestamp > Date.now() - 1000 * 60 * 60 * 24),
```
Comment (diversemix) [inline src/utils/searchCache.ts:37]: Reads a little "magical" - can we have a descriptive const for `1000 * 60 * 60 * 24` ?

**Review by diversemix — CHANGES_REQUESTED** [review]
Can we put the s3 path in as a config item so we're not hardcoded to staging?

File: src/components/blocks/Event.tsx (hunk 2 above)
Comment (PatFawbertMills) [reply src/components/blocks/Event.tsx:34]: Just old implementation when events had categories, e.g. target event. I'll remove for now

File: src/components/document/FamilyDocument.tsx (hunk 4 above)
Comment (PatFawbertMills) [reply src/components/document/FamilyDocument.tsx:53]: Yes we need to join up the thinking between the use of slugs on the site and supporting older slugs systematically.

**Review by diversemix — APPROVED** [review]
Looking good :man_dancing:

**Pull request merged**